plugins/generator-1.21.1/neoforge-1.21.1/templates/fluid/fluidtype.java.ftl

```diff
@@ -76,6 +80,38 @@ package ${package}.fluid.types;
 					return FLOWING_TEXTURE;
 				}
 
+				<#if data.textureRenderOverlay?has_content>
+				@Override public ResourceLocation getRenderOverlayTexture(Minecraft mc) {
+					return RENDER_OVERLAY_TEXTURE;
+				}
+				</#if>
+
+				<#if data.hasFog>
+					<#if data.fogColor?has_content>
+					@Override public Vector3f modifyFogColor(Camera camera, float partialTick, ClientLevel level, int renderDistance, float darkenWorldAmount, Vector3f fluidFogColor) {
+						return new Vector3f(${data.fogColor.getRed()/255}f, ${data.fogColor.getGreen()/255}f, ${data.fogColor.getBlue()/255}f);
+					}
+					</#if>
+
+					public void modifyFogRender(Camera camera, FogRenderer.FogMode mode, float renderDistance, float partialTick, float nearDistance, float farDistance, FogShape shape) {
```

Comment:
What would be the use case of fog start and end parameters, wouldn't one usually want to directly use nearDistance and farDistance parameters?

Reply:
There isn't really a standard vanilla behavior (powder snow and lava have priority over blindness, water doesn't), and custom fluids can only modify vanilla fog, so the fixed parameters help cover the basic cases. 
`nearDistance` and `farDistance` could be passed as dependencies though

Reply:
Yes, that may be a good idea. Marking as draft, mark back as RFR once added